src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Egress/S3/InMemoryS3.cs

```diff
@@ -0,0 +1,1042 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Net;
+using System.Threading;
+using Amazon.Runtime;
+using Amazon.S3;
+using Amazon.S3.Model;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests.Egress.S3
+{ 
+    public class InMemoryS3 : IAmazonS3
```

Comment:
Rather than reimplement the whole IAmazonS3 interface, consider creating a new interface that only contains the methods that the egress provider needs, have a default implementation that's used in dotnet-monitor, and the test version here.